base/constants_syncdocs.go

```diff
@@ -385,32 +385,45 @@ type SyncInfo struct {
 //  1. If syncInfo doesn't exist, it is created for the specified metadataID
 //  2. If syncInfo exists with a matching metadataID, returns requiresResync=false
 //  3. If syncInfo exists with a non-matching metadataID, returns requiresResync=true
-func InitSyncInfo(ds DataStore, metadataID string) (requiresResync bool, err error) {
+//     If syncInfo exists and has metaDataVersion greater than or equal to 4.0, return requiresAttachmentMigration=false, else requiresAttachmentMigration=true to bring migrate metadata attachments.
+func InitSyncInfo(ds DataStore, metadataID string) (requiresResync bool, requiresAttachmentMigration bool, err error) {
 
 	var syncInfo SyncInfo
 	_, fetchErr := ds.Get(SGSyncInfo, &syncInfo)
 	if IsDocNotFoundError(fetchErr) {
 		if metadataID == "" {
-			return false, nil
+			return false, true, nil
```

Comment:
I was running `go test -coverpkg ./... -cover=cover.out ./db -run TestSyncInfo` to see if lines were covered and I was surprised this line isn't hit.